trieste/models/optimizer.py

```diff
@@ -0,0 +1,118 @@
+from typing import Any, Dict, Optional, Callable, Union, Iterable, Tuple
+from functools import singledispatch
+from dataclasses import dataclass, field
+import tensorflow as tf
+import gpflow
+
+from gpflow.models import ExternalDataTrainingLossMixin, InternalDataTrainingLossMixin
+from ..utils import jit
+from ..datasets import Dataset
+
```

Comment:
there are a number of partial or missing type hints in this file. Do you mean to fill them in?

Reply:
Callable -> Callable[[], SOMETHING]